devito/symbolics/extended_sympy.py

```diff
@@ -398,6 +398,26 @@ def typ(self):
     @property
     def _op(self):
         return '(%s)' % self.typ
+    
+
+class FunctionPtr(UnaryOp):
+
+    """
+    Symbolic representation of C's function pointers.
+    """
+
+    _return_typ = ''
```

Comment:
why aren't these instance attributes?

also, you can add an 'e' at the end of 'type' ;)

Reply:
I took inspo from the Cast class https://github.com/devitocodes/devito/blob/0abe3952a792ed9ba5bf9ef904fa2fec04789a79/devito/symbolics/extended_sympy.py#L370C1-L400C33 and _base_typ is not an instance attribute. Is there a particular reason for this?

Reply:
I think it's a leftover from code that has been refactorer multiple times. You may probably tweak that too !